### PR TITLE
feat(camundaPlatform): add updateCamundaExclusiveBehavior

### LINKED ISSUE #29: Set `camunda:exlusive` flag to true, when not async anymore via custom behavior
__Is your feature request related to a problem? Please describe.__

When `camunda:asyncAfter` or `camunda:asyncBefore` are set to false, then a behavior should ensure, that the `camunda:exclusive` flag is set to true.

__Describe the solution you'd like__

Build camunda-platform behavior to set `camunda:exlusive` flag to true, when not async anymore.

This is to replace custom logic we previously had in the [`bpmn-js-properties-panel`](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/lib/provider/camunda/parts/implementation/AsyncContinuation.js#L96).

However, having this as a behavior is
* consistent to how we do it with other properties (compare zeebe propagateAllChildVariables)
* allows to have this logic effective not only when the properites panel is edited, but also when modeling happens via API

__Describe alternatives you've considered__

Implement in bpmn-js-properties-panel

__Additional context__

* Discussed: https://github.com/bpmn-io/bpmn-properties-panel/pull/39

### PR DESCRIPTION
**Pull request by MaxTru**
closes #29

### CODE REVIEW

File: lib/camunda-platform/features/modeling/behavior/UpdateCamundaExclusiveBehavior.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      // (1.1) Execute if...
+      if (is(element, 'camunda:AsyncCapable') && // ...asyncCapable
+      properties && // ...properties updated
+      (properties['camunda:asyncBefore'] === false || // ...update async (1)
+        properties['camunda:asyncAfter'] === false) && // ...update async (2)
+      isExclusive(businessObject) === false // ...is currently not exclusive
+      ) {
+
+        // (1.2) ... but don't execute if both async were set or one is set to true
+        if ((isAsyncAfter(businessObject) && isAsyncBefore(businessObject)) ||
```
Comment (pinussilvestrus) [inline lib/camunda-platform/features/modeling/behavior/UpdateCamundaExclusiveBehavior.js:42]: So it wouldn't execute when `asyncBefore=true` & `asyncAfter=true` _before_ the update? Can't it be the case when I update both properties in one command?

```js
{
  ...
  properties: {
    'camunda:asyncBefore': true,
    'camunda:asyncAfter': true
  }
}
```

That's surely not possible via properties panel, but maybe via manual command / `modeling.updateProperties`.

**Review by pinussilvestrus — COMMENTED** [review]
I pretty much like this approach 👍  One thing to consider.

File: lib/camunda-platform/features/modeling/behavior/UpdateCamundaExclusiveBehavior.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        properties
+      } = context;
+
+      const businessObject = getBusinessObject(element);
+
+      // (1.1) Execute if...
+      if (is(element, 'camunda:AsyncCapable') && // ...asyncCapable
+      properties && // ...properties updated
+      (properties['camunda:asyncBefore'] === false || // ...update async (1)
+        properties['camunda:asyncAfter'] === false) && // ...update async (2)
```
Comment (pinussilvestrus) [inline lib/camunda-platform/features/modeling/behavior/UpdateCamundaExclusiveBehavior.js:37]:
```suggestion
      (properties['camunda:asyncBefore'] === false || // ...update async before (1)
        properties['camunda:asyncAfter'] === false) && // ...update async after (2)
```

Comment (MaxTru) [reply lib/camunda-platform/features/modeling/behavior/UpdateCamundaExclusiveBehavior.js:37]: :+1:

File: lib/camunda-platform/features/modeling/behavior/UpdateCamundaExclusiveBehavior.js (hunk 1 above)
Comment (MaxTru) [reply lib/camunda-platform/features/modeling/behavior/UpdateCamundaExclusiveBehavior.js]: Ah true, I forgot that case. 

I added a test-case and adjusted the logic

Comment (MaxTru) [reply lib/camunda-platform/features/modeling/behavior/UpdateCamundaExclusiveBehavior.js]: ty

**Review by pinussilvestrus — APPROVED** [review]
Looks good👍

**Pull request merged**